### ISSUE TITLE
proposal: `type_init_super_parameters`

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#3210]
## Description

There is a [type_init_formals](https://dart-lang.github.io/linter/lints/type_init_formals.html) rule, which says "don't do `C(int this.f)`". Should we have a similar rule for "Don't do `C(int super.f)`".

## Kind

*Does this enforce style advice?  Guard against errors?  Other?* Consistency with `type_init_formals`?

## Good Examples 

```dart
class C {
  C(int f);
}
class D extends C {
  D(super.f);
}
```

## Bad Examples

```dart
class C {
  C(int f);
}
class D extends C {
  D(int super.f);
}
```

[Writing Lints]: https://github.com/dart-lang/linter/blob/master/doc/WritingLints.MD
[Effective Dart]: https://dart.dev/guides/language/effective-dart
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo]
[SDK Tracker]: https://github.com/dart-lang/sdk/issues

**Comment by eernstg** [comment]
I think `type_init_formals` and a new `type_super_parameters` should be treated similarly. However, `type_init_formals` is questionable, so I'd suggest that they both work a bit differently:

It makes sense to lint the case where an initializing formal has a declared type, and it is exactly the same type as the one that would have been inferred:

```dart
class A {
  int x;
  A(int this.x); // Lint, we'd get the same type by inference anyway.
}

class B extends A {
  B(int super.x); // Lint, we'd get the same type by inference anyway.
}
```

.. but it seems to be a false positive in the case where the types differ:

```dart
class Either {
  int? x, y; // Invariant: one of these is null, the other is non-null.

  // Public constructors enforce the invariant.
  Either.x(int this.x) : y = null;
  Either.y(int this.y) : x = null;

  // Private constructor, relies on correct invocations, but with assertion.
  Either._({this.x, this.y}) :
      assert((x != null && y == null) || (x == null && y != null));
}

class EitherDefault extends Either {
  EitherDefault.x({int super.x = 42}) : super._();
  EitherDefault.y({int super.y = 24}) : super._();
}
```

**Comment by srawlins** [comment]
I added a proposal to update Effective Dart: https://github.com/dart-lang/site-www/issues/4262
